### ISSUE TITLE
Add panel with events from culture.si

### ISSUE THREAD

**Issue by mitar** [wlanslovenija/PiplMesh#175]
Add panel with events from [culture.si](http://www.culture.si/). Add locations of those events to Google Maps panel (#173).

Define a background task to fetch regularly this data for all nodes (their locations) in the network.

If needed, work together with them to define a suitable API.

**Comment by agiz** [comment]
How accurate should this be? Is it ok to mark the city? If this is too wide can you provide an idea of locating "Kino Partizan" for example.

**Comment by mitar** [comment]
No. It is far from OK to mark the city. We need direct geo location of the venue.

And they have location displayed on the map, so they do have this information.

You should write to them.

**Comment by agiz** [comment]
They have some, but not all or maybe I am not seeing something. http://www.culture.si/en/Bled_Castle has location, http://www.culture.si/en/Klub_Zakon does not.

**Comment by mitar** [comment]
We will ignore those who do not have location. It should be in owner's interest to update its entry on culture.si.

**Comment by agiz** [comment]
There are literary no locations for any event listed in http://www.culture.si/en/Culture.si:Events. http://www.culture.si/en/Cankarjev_dom_Gallery is one rare exception. I do not know what to do here...

**Comment by mitar** [comment]
You have two options:
- you ignore those events for which venues are not known
- or you do reverse geocoding for as a fallback for those unknown venues from their address, Google has such an API, but we years ago when I was playing with it was not so good (got only the street right, the number was like mostly ignored), maybe it is better now

I see that they have updated [their API](https://developers.google.com/maps/documentation/geocoding/). Maybe you can play a bit with it and see if you can get from the result how precise it. Maybe we can try to do reverse geocoding and if the result is precise enough, we use it. Otherwise we ignore the venue.

(There is a third approach, which is a semi-commercial Slovenian database, which we could might "lend" to run through all addresses once and store that for later. Only new venues would then have problems.)

**Comment by agiz** [comment]
Almost all events fall under first option. We are left with options two and three. I think the third option will work best.

**Comment by mitar** [comment]
Please first try how good Google API is.

**Comment by mitar** [comment]
(We could always have it as a fallback from third option for new venues.)

**Comment by agiz** [comment]
I will, but this is another issue then, no? Reverse geocoding might be used in other issues as well.

**Comment by mitar** [comment]
You can do it here. And if there will be later on some other use too, we can extract it out.

**Comment by mitar** [comment]
One solution to problem could be that we retrieve list of all events from culture.si and for those events where there is no GPS coordinate, we do not put it on the map, but just list it in the list bellow with some additional link like "Can you help us place it on the map?" and if user clicks, she gets a big overlayed-over-content map where she can select a location. And we use this location.

Maybe similar could be done for all events, just that on other events we could have "Is there anything wrong with a position of the event?" or something like this.

Or this could be even a general interface to:
- fix existing events we have
- add your own events

**Comment by mitar** [comment]
I must say that I did a mistake here. I talked with guys from Ljudmila and the right page for Slovenian events is http://www.kulturnik.si/, not culture.si. kulturnik is an aggregator of existing content, so things are probably even worse to reuse, but we could try. :-) Anyway, they said that if need something, some feature or something, we should just ask.

**Comment by agiz** [comment]
I checked kulturnik.si. They do not have consistent representation of events, at least not those that are display on the webpage. For example there is no unified way to get location of the event. I will write them an email how to use their api.
